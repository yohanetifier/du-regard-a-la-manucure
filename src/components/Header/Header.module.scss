.header {
    display: flex;
    // justify-content: space-around;  
    align-items: center;    
    height: 100px; 
    width: 100vw; 
    position: fixed; 
    // border: 2px solid red; 
    // position: relative; 
}

.menu{
    margin: 0;    
    width: 50%;
    display: flex;     
    justify-content: space-between;
    align-items: center;     
    & > a:last-child {
        display: flex; 
        justify-content:center;
        align-items: center; 
        text-align: center; 
        border: 2px solid pink; 
        height: 30px; 
        border-radius: 5px;
        transition: all 500ms 2ms; 
        &:hover {
        background-color: pink;
        color: white; 
    }
}
}

.link { 
    text-align: center; 
    // border: 2px solid red; 
    text-decoration: none;
    color: black; 
    font-weight: bold;
    width: 30%; 
    height: 100%; 
}

.img{
    margin-left: 100px; 
    // border: 2px solid red; 
    height:100%; 
}

.logo {
    // border: 2px solid red; 
    display: flex; 
    align-items: center; 
    width: 30%; 
    height: 100%; 
}

.menuhidden {
    border: 2px solid red; 
    width: 100vw; 
    height: 30vh; 
    position: absolute; 
    background-color: transparent; 
}


