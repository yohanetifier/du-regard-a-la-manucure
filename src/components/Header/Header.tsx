import styles from './Header.module.scss';
import { Link } from 'react-router-dom';
import logo from '../../assets/images/logo/V1_Rose.svg';
// import logo from '../../assets/images/logo/du_regard_a_la_manucure.svg'
import { useState, useContext } from 'react';
import { motion } from 'framer-motion';
import Image from '../Image/Image';
import SocialNetwork from '../SocialNetwork/SocialNetwork';
import { BackgroundColor } from '../App/App';

export interface Navigation {
    label: string
    to: string
}

const navigation: Navigation[] = [
    {
        label: 'ACCUEIL',
        to: '/'
    },
    {
        label: 'A PROPOS',
        to: '/about'
    },
    {
        label: 'NOS PRESTATIONS',
        to: '/prestations'
    },
    {
        label: 'NOS SERVICES',
        to: '/services'
    },
    {
        label: 'CONTACT',
        to: '/contact'
    },
    {
        label: 'MENTIONS LEGALES',
        to: '/mentions-legales'
    },

]

const container = {
    show: {
        transition: {
            staggerChildren: 0.2
        }
    }

}

const item = {
    hidden: {
        opacity: 0,
    },
    show: {
        opacity: 1,
    }
}

const Header = () => {
    const [menu, setMenu] = useState(false);
    const { isBackgroundColorIsPink } = useContext(BackgroundColor);

    const changeColor = {
        purple: {
            filter: 'invert(7%) sepia(85%) saturate(6792%) hue-rotate(334deg) brightness(103%) contrast(89%)',
            transition: 'filter 200ms'
        },
        black: {
            fill: 'black',
            transition: 'fill 200ms'
        }
    }

    const { purple } = changeColor;
    const { black } = changeColor;

    return (
        <div className={styles.header}>
            <Link className={styles.logo} to='/'>
                <img src={logo} alt="" className={styles.img} style={!menu ? purple : black} />
            </Link>
            <div className={styles.menu}>
                <div className={styles.menulayout} onClick={() => !menu ? setMenu(true) : setMenu(false)}>
                    <span className={!menu ? styles.firstline : `${styles.firstline} ${styles.movefirstline}`}></span>
                    <span className={!menu ? styles.secondline : `${styles.secondline} ${styles.movesecondline}`}></span>
                    <span className={!menu ? styles.thirdline : `${styles.thirdline} ${styles.movethirdline}`}></span>
                </div>
            </div>
            <div className={menu ? `${styles.inactifmenu} ${styles.actifmenu}` : `${styles.inactifmenu}`}>
                {menu && <motion.ul
                    className={styles.wrapperlist}
                    variants={container}
                    initial='hidden'
                    animate='show'
                >
                    {navigation.map(({ label, to }) => (
                        <motion.li variants={item} className={styles.list}><Link to={to} className={styles.link} onClick={() => setMenu(false)}>{label}</Link></motion.li>
                    ))}
                </motion.ul>}
            </div>
            <SocialNetwork
                className={styles.socialNetwork}
                menu={menu}
            />
        </div>
    )
}

export default Header;

