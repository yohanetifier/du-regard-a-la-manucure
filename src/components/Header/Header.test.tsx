import { shallow, ShallowWrapper } from 'enzyme'
import Header from './Header'
import Enzyme from 'enzyme'
import toJson from "enzyme-to-json";
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

let wrapper: ShallowWrapper

beforeEach(() => {
    wrapper = shallow(<Header />)
})

describe('Header', () => {
    it('should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('should be a div', () => {
        expect(wrapper.type()).toBe('div')
    })
    it('should have a classname of header', () => {
        expect(wrapper.prop('className')).toBe('header')
    })
    it('should have a div as firstchild', () => {
        expect(wrapper.childAt(0).type()).toBe('div')
    })
    it('the firstchild have the class logo', () => {
        expect(wrapper.childAt(0).prop('className')).toBe('logo')
    })
    it('have a h1 as child', () => {
        const firstChild = wrapper.childAt(0)
        expect(firstChild.childAt(0).type()).toBe('h1')
    })
    it('should display the name of the entreprise', () => {
        expect(wrapper.childAt(0).text()).toBe('Du regard Ã  la manucure')
    })
    it('should have a ul as second child', () => {
        expect(wrapper.childAt(1).type()).toBe('ul')
    })
    it('the second child should have the class of menu', () => {
        expect(wrapper.childAt(1).prop('className')).toBe('menu')
    })
    it('should have 4 road as child', () => {
        expect(wrapper.childAt(1).children()).toHaveLength(4)
    })
    it('should have a Home Road', () => {
        const firstRoad = wrapper.childAt(1).childAt(0)
        expect(firstRoad.prop("className")).toBe('link')
        expect(firstRoad.prop("to")).toBe('/home')
        expect(firstRoad.text()).toBe('Home')
    })
    it('should have a Services Road', () => {
        const firstRoad = wrapper.childAt(1).childAt(1)
        expect(firstRoad.prop("className")).toBe('link')
        expect(firstRoad.prop("to")).toBe('/services')
        expect(firstRoad.text()).toBe('Services')
    })
    it('should have a Contact Road', () => {
        const firstRoad = wrapper.childAt(1).childAt(2)
        expect(firstRoad.prop("className")).toBe('link')
        expect(firstRoad.prop("to")).toBe('/contact')
        expect(firstRoad.text()).toBe('Contact')
    })
    it('should have a Reservation Road', () => {
        const firstRoad = wrapper.childAt(1).childAt(3)
        expect(firstRoad.prop("className")).toBe('link')
        expect(firstRoad.prop("to")).toBe('/rendez-vous')
        expect(firstRoad.text()).toBe('Reservez')
    })
    

})