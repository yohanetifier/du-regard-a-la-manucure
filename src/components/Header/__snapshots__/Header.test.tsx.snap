// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="header"
>
  <div
    className="logo"
  >
    <h1
      className="title"
    >
      Du regard Ã  la manucure
    </h1>
  </div>
  <ul
    className="menu"
  >
    <Link
      className="link"
      key="0"
      to="/"
    >
      Home
    </Link>
    <Link
      className="link"
      key="1"
      to="/services"
    >
      Services
    </Link>
    <Link
      className="link"
      key="2"
      to="/contact"
    >
      Contact
    </Link>
    <Link
      className="link"
      key="3"
      to="/rendez-vous"
    >
      Reservez
    </Link>
  </ul>
</div>
`;
